### PR TITLE
Robot factory was created

### PR DESCRIPTION
**Pull request by serge-kontakt**

### CODE REVIEW

File: src/robotFactory.js [hunk 1]
```diff
@@ -1,10 +1,85 @@
...
+    this.weight = weight;
+    this.chipVersion = chipVersion;
+    this.coords = coords;
 
-class FlyingRobot {}
+    if (!coords.x) {
+      this.coords.x = 0;
+    } else {
+      this.coords.x = coords.x;
+    }
```
Comment (Anastasiia-Svintsova) [inline src/robotFactory.js:14]:
```suggestion
    this.coords.x = coords.x || 0;
```

File: src/robotFactory.js [hunk 2]
```diff
@@ -1,10 +1,85 @@
...
 
-class DeliveryDrone {}
+    if (!coords.y) {
+      this.coords.y = 0;
+    } else {
+      this.coords.y = coords.y;
+    }
+  }
```
Comment (Anastasiia-Svintsova) [inline src/robotFactory.js:21]: Same

File: src/robotFactory.js [hunk 3]
```diff
@@ -1,10 +1,85 @@
...
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    if (!coords.z) {
+      this.coords.z = 0;
+    } else {
+      this.coords.z = coords.z;
+    }
```
Comment (Anastasiia-Svintsova) [inline src/robotFactory.js:53]: same

File: src/robotFactory.js [hunk 4]
```diff
@@ -1,10 +1,85 @@
...
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion,
+    maxLoadWeight, currentLoad = null) {
```
Comment (Anastasiia-Svintsova) [inline src/robotFactory.js:67]: No need to add a default value to currentLoad
```suggestion
  constructor(
    name, 
    weight, 
    coords, 
    chipVersion,
    maxLoadWeight, 
    currentLoad
  ) {
```

File: src/robotFactory.js [hunk 5]
```diff
@@ -1,10 +1,85 @@
...
+    super(name, weight, coords, chipVersion);
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (this.currentLoad === null && cargo.weight <= this.maxLoadWeight) {
```
Comment (Anastasiia-Svintsova) [inline src/robotFactory.js:74]:
```suggestion
    if (!this.currentLoad && cargo.weight <= this.maxLoadWeight) {
```

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]
Good! Let's improve the code a little bit

File: src/robotFactory.js (hunk 4 above)
Comment (serge-kontakt) [reply src/robotFactory.js:52]: Without default value the test is failed - screenshot.
Снимок экрана 2023-02-27 в 15 48 28

**Pull request closed without merging**